File: daily_python/2678_Number_of_Senior_Citizens.py
import os
import sys
from typing import List

class Solution:
    def countSeniors(self, details: List[str]) -> int:
        ans = 0
        for passenger in details:
            if int(passenger[11:13]) > 60:
                ans += 1
        return ans
        
def readData(filename):
    dataset = []
    with open(filename, 'r') as file:
        context = file.read().strip()
        datas = context.split('\n')
        for lines in datas:
            line = lines.split('=')[1].strip()[1:-2].replace('\"',"")
            data = line.split(',')
            dataset.append(data)
    return dataset


if __name__ == '__main__':
    if (len(sys.argv)) == 1:
        path = os.path.basename(__file__)
        filename = os.path.splitext(path)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readData(filename)
    solution = Solution()
    results = []
    for data in dataset:
        results.append(solution.countSeniors(data))
    for index, result in enumerate(results):
        print(f'data {index+1} : {result}')

File: daily/2024/2678_Number_of_Senior_Citizens.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int countSeniors(vector<string>& details) {
        int ans = 0;
        for (string passenger : details) if ((passenger[11] - '0') * 10 + (passenger[12] - '0') > 60) ans++;
        return ans;
    }
};

int main(){
    vector<string> details = {"7868190130M7522","5303914400F9211","9273338290F4010"};
    // vector<string> details = {"1313579440F2036","2921522980M5644"};
    Solution S;
    cout << S.countSeniors(details) << endl;
    return 0;
}